corgi/core/constants.py

```diff
@@ -193,3 +193,94 @@
   END;
   $$ LANGUAGE plpgsql IMMUTABLE PARALLEL SAFE;
 """
+
+GET_LATEST_COMPONENTS_STOREDPROC_SQL = """
+CREATE OR REPLACE FUNCTION get_latest_components( model_type text, ofuris text[], component_type text, component_ns text, component_name text, component_arch text, include_inactive_streams boolean) RETURNS SETOF uuid AS $$
+  DECLARE
+    p_ofuri text;
+    component_uuid text;
+    component_epoch int;
+    component_version text;
+    component_release text;
+    latest_uuid text;
+    latest_epoch int;
+    latest_version text;
+    latest_release text;
+
+    product_component_cursor CURSOR FOR
```

Comment:
Did you consider calling the get_latest_component function instead of redefining it here? I think it would help reduce the burden of maintaining this code into the future. Can't say or not if this is actually exactly redefining that function, but it does look similar. If we can't call the database function here perhaps it is a good idea to make use of the Python constants so we don't have update both database function if the logic changes.

Reply:
the idea is that **get_latest_components** should 'go away' when we do graphdb refactor ...